website/src/components/Chat/ChatListItem.tsx

```diff
@@ -1,41 +1,119 @@
-import { Box, Button, Flex, Tooltip, useToast } from "@chakra-ui/react";
-import { LucideIcon, Pencil, Trash2 } from "lucide-react";
+import { Box, Button, CircularProgress, Flex, Input, Tooltip, useBoolean, useOutsideClick } from "@chakra-ui/react";
+import { Check, LucideIcon, Pencil, Trash2, X } from "lucide-react";
 import Link from "next/link";
 import { useRouter } from "next/router";
 import { useTranslation } from "next-i18next";
-import { MouseEvent, useCallback } from "react";
-import { del } from "src/lib/api";
-import { ROUTES } from "src/lib/routes";
+import { KeyboardEvent, MouseEvent, useCallback, useRef } from "react";
+import { del, put } from "src/lib/api";
+import { API_ROUTES, ROUTES } from "src/lib/routes";
 import { ChatItem } from "src/types/Chat";
 import useSWRMutation from "swr/mutation";
 
-export const ChatListItem = ({ chat }: { chat: ChatItem }) => {
+export const ChatListItem = ({
+  chat,
+  onUpdateTitle,
+}: {
+  chat: ChatItem;
+  onUpdateTitle: (params: { chatId: string; title: string }) => void;
+}) => {
   const { query } = useRouter();
   const { t } = useTranslation("chat");
   const isActive = chat.id === query.id;
+  const [isEditing, setIsEditing] = useBoolean(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const rootRef = useRef<HTMLDivElement>(null);
+  useOutsideClick({
+    ref: rootRef,
+    handler: () => {
+      if (isEditing) {
+        setIsEditing.off();
+      }
+    },
+  });
+  const { trigger: updateChatTitle, isMutating: isUpdatingTitle } = useSWRMutation(
+    API_ROUTES.UPDATE_CHAT_TITLE(chat.id),
+    put
+  );
+  const handleConfirmEdit = useCallback(async () => {
+    const title = inputRef.current?.value;
+    if (!title) return;
+    await updateChatTitle({ title, chat_id: chat.id });
+    setIsEditing.off();
+    onUpdateTitle({ chatId: chat.id, title });
+  }, [chat.id, onUpdateTitle, setIsEditing, updateChatTitle]);
+
+  const handleKeyDown = useCallback(
+    (e: KeyboardEvent) => {
+      if (e.key === "Escape") {
+        setIsEditing.off();
+      }
+      if (e.key === "Enter") {
+        handleConfirmEdit();
+      }
+    },
+    [handleConfirmEdit, setIsEditing]
+  );
+
   return (
     <Button
-      as={Link}
-      href={ROUTES.CHAT(chat.id)}
+      // @ts-expect-error error due to dynamicly changing as prop
+      ref={rootRef}
+      {...(!isEditing ? { as: Link, href: ROUTES.CHAT(chat.id) } : { as: "div" })}
```

Comment:
Does this work? keep it as `Link` & remove the href when editing (or set to '#'), then it would not navigate.